src/Units/UnitModel.java

```diff
@@ -123,9 +116,9 @@ private UnitModel(String pName, UnitRoleEnum pRole, ChassisEnum pChassis, int co
 
     for( TerrainType terrain : TerrainType.TerrainTypeList )
     {
-      if( (((chassis == ChassisEnum.AIR_HIGH) || (chassis == ChassisEnum.AIR_LOW)) && terrain.healsAir()) ||
-          (((chassis == ChassisEnum.TANK) || (chassis == ChassisEnum.TROOP)) && terrain.healsLand()) ||
-          (((chassis == ChassisEnum.SHIP) || (chassis == ChassisEnum.SUBMERGED)) && terrain.healsSea()) )
+      if( (((role & (AIR_LOW | AIR_HIGH)) > 0) && terrain.healsAir()) ||
```

Comment:
We should put these kinds of messy comparisons in a function; `hasRole(AIR_LOW | AIR_HIGH)` or some such.